daemon/src/wallet.rs

```diff
@@ -0,0 +1,50 @@
+use crate::model::{WalletError, WalletInfo};
+use anyhow::Result;
+use bdk::bitcoin::Amount;
+use bdk::blockchain::{Blockchain, NoopProgress};
+use bdk::database::BatchDatabase;
+use bdk::wallet::AddressIndex;
+use tokio::sync::watch;
+
+pub fn wallet_sync<B, D>(
+    wallet_feed_sender: &watch::Sender<WalletInfo>,
+    wallet: &bdk::Wallet<B, D>,
+    previous: WalletInfo,
+) -> Result<Option<WalletInfo>, WalletError>
+where
+    B: Blockchain,
+    D: BatchDatabase,
+{
+    wallet
+        .sync(NoopProgress, None)
+        .map_err(|_| WalletError::Sync)?;
```

Comment:
Technically, this could take a very long time and be blocking. We might want to put this on tokio's blocking task pool!

https://docs.rs/tokio/1.11.0/tokio/task/fn.spawn_blocking.html

Reply:
I am actually not sure in what scenario it would take long. I think it would only take long if someone uses an existing wallet. If we assume that the tool will always generate a wallet and one has to transfer funds into it it won't make much time. 
Additionally there is the initial sync when the daemon starts up, which can take longer (the user will see that it takes long...)
Still happy to give tokio blocking task pools a try, but I think it's actually not a must atm.

Reply:
Your code here is generic over the wallet backend. If it is a p2p backend, you need to download all blocks since you were last online, that can take hours!

If you want to make the assumption that sync doesn't take long because we are using esplora, then don't be generic over the backend :)

If you want to be generic over the backend, write the code in a way so it works with all backends!